src/structlog/stdlib.py

```diff
@@ -451,7 +451,11 @@ def format(self, record):
         # Make a shallow copy of the record to let other handlers/formatters
         # process the original one
         record = logging.makeLogRecord(record.__dict__)
-        if isinstance(record.msg, dict):
+        if (
+            isinstance(record.msg, dict)
+            and hasattr(record, "_logger")
```

Comment:
Please meet: https://hynek.me/articles/hasattr/ :)

If we pile on checks and assume that dicts are predominant, why don't we invert the whole thing and make it an `try:…except (KeyError, AttributeError)`? I reckon that’s considered more pythonic too.

Reply:
I would argue when you have the ProcessorLogger in the mix, dicts and strs aren't predominant one way or the other, so a try...except seems unnecessarily slow.

I wonder if instead the wrapper should convert LogRecord into WrappedLogRecord and just do a type-check since ultimately what we're looking for is a different object type.

Reply:
>  try...except seems unnecessarily slow.

Hm, do you really think it's slower than multiple checks on each record? I think I'd like to see a benchmark on that since this is is hot code. If you have some spare it would be great if you could compare it using https://pypi.org/project/perf/ . I only care about modern Python versions FWIW.

> I wonder if instead the wrapper should convert LogRecord into WrappedLogRecord and just do a type-check since ultimately what we're looking for is a different object type.

That sound clean, but really slow? 🤔

Sorry I'm being so anal about performance here but I take great pride that structlog isn't a handbrake like logging and I'd like it to stay that way. 😇

Reply:
No I agree about keeping it quick. I'll setup some tests.

Reply:
Awesome thank you and sorry for the delay. My original plan was to run them myself but just didn't get around to it.